dodola/tests/test_services.py

```diff
@@ -636,6 +610,131 @@ def test_aiqpd_train(tmpdir, monkeypatch):
     assert aiqpd_model.af.shape == af_expected_shape
 
 
+def test_aiqpd_integration(tmpdir, monkeypatch):
+    """Tests that the shape of adjustment factors matches the expected shape"""
+    monkeypatch.setenv(
+        "HDF5_USE_FILE_LOCKING", "FALSE"
+    )  # Avoid thread lock conflicts with dask scheduler
+    lon = [-99.83, -99.32, -99.79, -99.23]
+    lat = [42.25, 42.21, 42.63, 42.59]
+    time = pd.date_range(start="1994-12-17", end="2015-01-15")
+    temperature_ref = 15 + 8 * np.random.randn(len(time), 4, 4)
+    temperature_train = 15 + 8 * np.random.randn(len(time), 4, 4)
+    variable = "scen"
+
+    ref_fine = xr.Dataset(
+        data_vars=dict(
+            scen=(["time", "lat", "lon"], temperature_ref),
+        ),
+        coords=dict(
+            time=time,
+            lon=(["lon"], lon),
+            lat=(["lat"], lat),
+        ),
+        attrs=dict(description="Weather related data."),
+    )
+
+    ds_train = xr.Dataset(
+        data_vars=dict(
+            scen=(["time", "lat", "lon"], temperature_train),
+        ),
+        coords=dict(
+            time=time,
+            lon=(["lon"], lon),
+            lat=(["lat"], lat),
+        ),
+        attrs=dict(description="Weather related data."),
+    )
+
+    # remove leap days
+    ref_fine = convert_calendar(ref_fine[variable], target="noleap")
+    ds_train = convert_calendar(ds_train[variable], target="noleap")
+
+    # take the mean across space to represent coarse reference data for AFs
+    ds_ref_coarse = ref_fine.mean(["lat", "lon"])
+    ds_train = ds_train.mean(["lat", "lon"])
+    ref_coarse = ds_ref_coarse.broadcast_like(ref_fine)
```

Comment:
Suggest adding a comment explaining why you're broadcasting this. It wasn't readily apparent to me.

Reply:
Done, thanks for the suggestion.